### DISCUSSION TITLE
Train Custom Text Recognition Model For All Indian Languages.

### DISCUSSION THREAD

**Discussion by chintan-eonian** [mindee/doctr#1343] [Q&A] [answered]
Hey, Newbie (in ML) here. 
First of all thanks for developing an open source project. 

I just wanted to ask,
**1) how much Dataset do I need for training a model for a new language?
2) and the example provided in docs for images as shown below, is there any online platform or a way to generate these type of images for a language?** 

labels.json
{
    "img_1.jpg": "I",
    "img_2.jpg": "am",
    "img_3.jpg": "a",
    "img_4.jpg": "Jedi",
    "img_5.jpg": "!",
    ...
}

**3) are there any constraints regarding the images? (like the size or resolution)**

**Comment by felixT2K** [comment, marked as answer]
Hi @chintan-eonian 👋🏼,

1. The current models are pretrained on ~11M real world image word crops (french vocab characters), so for non latin i suggest to train the model from scratch.
A good starting point would be 5M + (depending a bit on the character distribution in the dataset)

2. Generators you could try:
https://github.com/clovaai/synthtiger
https://github.com/PaddlePaddle/PaddleOCR/tree/release/2.7/StyleText

3. All images will be reshaped to a size of 32x128 i highly suggest if you use one of the above generators to generate differently sized images. 
Furthermore we apply already some augmentations so there is no need to apply any "hard" transformations in front.
ref.: https://github.com/mindee/doctr/blob/50d65d79c115cd6046b0ae6dfd61fbe061512a42/references/recognition/train_pytorch.py#L283

**Reply by chintan-eonian** [reply]
"5M" as in 5 Million right? 
there must be a way to automate this process (generating images) right? given the link you provided above for Generators?

**Reply by felixT2K** [reply]
Yes and Yes :) 
You can read a bit in the linked repositories and try to generate some data :) 
Feel free to share some generated samples that we can have a look 👍🏼

**Reply by chintan-eonian** [reply]
Hey read into it, thank you sooo much. <3 
and also one last question (sorry I forgot to add)..
as I have said I am new to ML, I wanted to ask, for training the model on this particular data, how much processing power do I need?

**Reply by felixT2K** [reply]
You need a nvidia gpu (best with > 8GB VRAM), 8GB RAM should be fine and a actual normal CPU :)

**Reply by chintan-eonian** [reply]
alright brother. thank you so much for quick response :)
